conformance/tests/httproute-invalid-reference-policy.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteInvalidReferencePolicy)
+}
+
+var HTTPRouteInvalidReferencePolicy = suite.ConformanceTest{
+	ShortName:   "HTTPRouteInvalidReferencePolicy",
+	Description: "A single HTTPRoute in the gateway-conformance-infra namespace should fail to attach to a Gateway in the same namespace if the route has a backendRef Service in the gateway-conformance-app-backend namespace and a ReferencePolicy exists but does not grant permission to route to that specific Service",
+	Manifests:   []string{"tests/httproute-invalid-reference-policy.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+		if !slices.Contains(s.ExtendedSupport, suite.SupportReferencePolicy) {
+			t.Skip("Skipping ReferencePolicy conformance test")
+		}
+
+		routeNN := types.NamespacedName{Name: "invalid-reference-policy", Namespace: "gateway-conformance-infra"}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: "gateway-conformance-infra"}
+
+		ns := v1alpha2.Namespace(gwNN.Namespace)
+		kind := v1alpha2.Kind("Gateway")
+
+		t.Run("Route status should have a route parent status with an Accepted condition set to False", func(t *testing.T) {
+			parents := []v1alpha2.RouteParentStatus{{
+				ParentRef: v1alpha2.ParentReference{
+					Group:     (*v1alpha2.Group)(&v1alpha2.GroupVersion.Group),
+					Kind:      &kind,
+					Name:      v1alpha2.ObjectName(gwNN.Name),
+					Namespace: &ns,
```

Comment:
The namespace assertion won't work here for our implementation unless the `HTTPRoute` includes the namespace on its `parentRefs` over in `httproute-invalid-reference-policy.yaml`. I'm not sure if this is how it's supposed to work - in which case simply adding the namespace to `parentRefs[0]` has us passing these tests - or if we should be implicitly adding the `HTTPRoute`'s own namespace to the parent ref if not specified.

Can we look into this together later today or next week?

Reply:
Pushed up a commit which I hope should fix this - I believe passing `false` as the `namespaceRequired` argument to `HTTPRouteMustHaveParents` should allow this field to be omitted from the route config (with the prior assumption that the route and expected parent being checked are in the same namespace).

I think there's a bit of an edge case that this boolean flag lets slip through in that it wouldn't be able to properly validate a route with multiple parents where one parent is in the same namespace (so namespace on the status could be optional), but a second parent in a different namespace, where the namespace would be required - I don't think we have any tests exercising a scenario like that yet though, so maybe something to improve as we look into adding route delegation conformance tests.

Reply:
I'm highly in favor of clarifying optional fields as much we possibly can in the spec. IMO they represent the biggest risk of divergence between implementations.